povertymapping/hdx.py

```diff
@@ -210,4 +209,32 @@ def get_hdx_file(
 
     logger.info(f"HDX Data: Successfully downloaded and cached for {region} at {zipfile_path}!")
 
-    return unzipped_hdxfile
\ No newline at end of file
+    return unzipped_hdxfile
+
+
+def generate_hdx_features(
+    aoi: pd.DataFrame,
+    region: str,
+    extra_args: dict = dict(nodata=np.nan),
```

Comment:
might be better to use `None` as the default argument for `extra_args` instead of a dictionary. Using mutable Python objects as a default value for Python functions (dicts and lists are mutable) can lead to unexpected bugs later down the line lol


More detail described [in this blog post](https://docs.python-guide.org/writing/gotchas/), and [here](https://www.geeksforgeeks.org/use-mutable-default-value-as-an-argument-in-python/). Recommended approach is to use `None` in the default value an then define the mutable object (in this case dictionary) within the function

Reply:
Interesting. I thought that all variables in functions are local but that doesn't seem to be the case for mutable default arguments.

I copied this default arg from `compute_raster_stats()` of `rollout_grids.py` so should I edit that as well, right? Noting here other cases with `list` or `dict` as default args:

`dhs.py`
- `DHSDataManager.__init__()`

`rollout_grids.py`
- `get_region_filtered_bingtile_grids()`
- `generate_dhs_cluster_level_data()`
- `generate_dhs_household_level_data()`
- `generate_dhs_cluster_level_data()`

`nightlights.py`
- `generate_nightlights_feature()`

Reply:
> I thought that all variables in functions are local but that doesn't seem to be the case for mutable default arguments.

Yeah it is still local! There's a slight nuance hehe. If it's defined within the function then it's local to the function. The `extra_args` is considered local, the mutable object was "created" globally and persists globally. 

Very slight nuances in mutating a variable vs assigning a variable sample here from a [reddit thread](https://www.reddit.com/r/learnpython/comments/wntguq/why_are_only_mutable_variables_a_problem_in/)
```
def func(a=[]):
a = a + [3]
return a

func()
> [3]
func()
> [3]

def func(a=[]):
a += [3]
return a

func()
> [3]
func()
> [3, 3]
```

Reply:
Although I also just realized maybe just using `None`  is not the straightforward solution since there could be multiple other arguments haha

Reply:
It ok to use a dictionary as an input parameter as long as the default value is `None`, something like 
```
---
extra_args: dict | None = None
): 
if extra_args = None:
   extra_args = dict(nodata=np.nan)
```

Reply:
hmmm regarding the other functions that also use a dictionary as a default value, I think would be good to also modify! But lmk if modifying will be cumbersome, in that case we can file an issue and address it in another PR instead

Reply:
Went over them and the fix is simple enough for the 7 functions (`DHSDataManager.__init__()` is a false alarm) so can do it in this PR.